arch: arm: soc: add STM32L471XG series

kzyapkov opened the pull request:
Similar to #9110 and #8772, this PR adds support for STM32L471XG.

erwango reviewed (changes requested):
Some minor comments.
Last, if stm32l475 is a superset of stm32l471, can't we have stm32l475.dtsi including stm32l471.dtsi ?

kzyapkov commented on dts/arm/st/stm32l471.dtsi:
```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2018 Allterco Robotics
```
done

kzyapkov commented on arch/arm/soc/st_stm32/stm32l4/Kconfig.defconfig.stm32l471xx:
```diff
@@ -0,0 +1,39 @@
+#
+# Copyright (c) 2016 Open-RnD Sp. z o.o.
+# Copyright (c) 2016 BayLibre, SAS
+# Copyright (c) 2018 Allterco Robotics
```
done

kzyapkov commented:
l745 is not a superset of l741, rather, the two have a common subset. Differences are:
 * L475 has usb, L741 does not
 * on L471 timers 6 and 7 are on APB1, while L475 has timers 8 and 17 on APB2

I don't know how to do that other than add a new file with common peripherals for the two to include.

I looked at the [STM32L4x1 reference manual](https://www.st.com/resource/en/reference_manual/dm00149427.pdf), but please double-check.

Commit:
soc: st: add STM32L471xG

This commit adds support for STM32L471xG. Since STM32L475 only adds
USB, stm32l475.dtsi now includes stm32l471.dtsi

Signed-off-by: Kiril Zyapkov <k.zyapkov@allterco.com>

kzyapkov commented:
Sorry, I never managed to get around to checking these registers. Will do shortly.

kzyapkov commented:
@erwango adapted the change to new board layout and reworked dts as you recommended.

erwango commented on soc/arm/st_stm32/stm32l4/Kconfig.defconfig.stm32l471xx:
```diff
@@ -0,0 +1,37 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if SOC_STM32L471XG
```
Can you rename to SOC_STM32L471X**X**?
Package information (last letter of the reference name) is not needed anymore since we extract package sizes from dts

erwango reviewed (changes requested):
Last request + a rebase and I hope we'll be good

kzyapkov commented:
Done. `stm32l471.dtsi` includes the latest updates from `stm32l475.dtsi` and adds `i2c3`.

kzyapkov commented:
Another rebase + a bunch of minor fixes.
